### PR TITLE
[#1423] added more generic http status code handling for frontend and…

### PR DESCRIPTION
**Pull request by Skylsmoi**
… apps + added better error msg for unhandled error

https://github.com/tracim/tracim/issues/1423

### CODE REVIEW

File: frontend_lib/src/helper.js [hunk 1]
```diff
@@ -13,25 +16,33 @@ const generateFetchResponse = async fetchResult => {
...
+  if (status === 204) return fetchResult // no result
+  if (status >= 200 && status <= 299) return generateFetchResponse(fetchResult)
+  if (status >= 300 && status <= 399) return generateFetchResponse(fetchResult)
+  if (status === 401) { // unauthorized
+    GLOBAL_dispatchEvent({type: 'disconnectedFromApi', date: {}})
+    return generateFetchResponse(fetchResult)
+  }
+  if (status >= 400 && status <= 499) return generateFetchResponse(fetchResult) // let specific handler handle it with fetchResult.body.code
+  if (status >= 500 && status <= 599) {
+    GLOBAL_dispatchEvent(errorFlashMessageTemplateObject(`Unexpected api error with http status ${status}`)) // no need for translation
```
Comment (buxx) [inline frontend_lib/src/helper.js]: There is no traduction here ?

File: frontend_lib/src/helper.js [hunk 2]
```diff
@@ -13,25 +16,33 @@ const generateFetchResponse = async fetchResult => {
...
+    return generateFetchResponse(fetchResult)
   }
+
+  GLOBAL_dispatchEvent(errorFlashMessageTemplateObject(`Unknown http status ${status}`)) // no need for translation
```
Comment (buxx) [inline frontend_lib/src/helper.js]: There is no traduction here ?

File: frontend_lib/src/helper.js (hunk 1 above)
Comment (buxx) [reply frontend_lib/src/helper.js]: This is the case where we decide to keep it in english maybe ?

File: frontend_lib/src/helper.js (hunk 2 above)
Comment (buxx) [reply frontend_lib/src/helper.js]: This is the case where we decide to keep it in english maybe ?

Comment (Skylsmoi) [reply frontend_lib/src/helper.js]: it is

File: frontend_lib/src/helper.js (hunk 1 above)
Comment (buxx) [reply frontend_lib/src/helper.js]: if yes, you can consider this MR have my approbation.

File: frontend_lib/src/helper.js (hunk 2 above)
Comment (buxx) [reply frontend_lib/src/helper.js]: if yes, you can consider this MR have my approbation.

File: frontend_lib/src/helper.js (hunk 1 above)
Comment (Skylsmoi) [reply frontend_lib/src/helper.js]: it is

**Pull request merged**